x/nexus/exported/types.go

```diff
@@ -258,20 +254,16 @@ func NewGeneralMessage(id string, sourceChain ChainName, sender string, destChai
 
 // ValidateBasic validates the general message
 func (m GeneralMessage) ValidateBasic() error {
-	if err := m.ID.ValidateBasic(); err != nil {
-		return sdkerrors.Wrap(err, "invalid id")
+	if err := utils.ValidateString(m.ID); err != nil {
+		return sdkerrors.Wrap(err, "invalid general message id")
 	}
 
-	if err := m.SourceChain.Validate(); err != nil {
+	if err := m.Sender.Validate(); err != nil {
 		return sdkerrors.Wrap(err, "invalid source chain")
 	}
 
-	if err := utils.ValidateString(m.Sender); err != nil {
-		return sdkerrors.Wrap(err, "invalid sender")
-	}
-
-	if err := utils.ValidateString(m.Receiver); err != nil {
-		return sdkerrors.Wrap(err, "invalid receiver")
+	if err := m.Recipient.Validate(); err != nil {
+		return sdkerrors.Wrap(err, "invalid source chain")
```

Comment:
```suggestion
		return sdkerrors.Wrap(err, "invalid destination chain")
```
or maybe just "invalid destination"